examples/hotrod/pkg/tracing/rpcmetrics/observer_test.go

```diff
@@ -152,9 +142,20 @@ func TestTags(t *testing.T) {
 		}
 		t.Run(fmt.Sprintf("%s-%v-%s", testCase.key, testCase.value, testCase.variant), func(t *testing.T) {
 			withTestTracer(func(testTracer *testTracer) {
-				span := testTracer.tracer.StartSpan("span", ext.SpanKindRPCServer)
-				span.SetTag(testCase.key, testCase.value)
-				span.Finish()
+				_, span := testTracer.tracer.Start(context.Background(), "span", trace.WithSpanKind(trace.SpanKindServer))
+				switch v := testCase.value.(type) {
```

Comment:
the only values in the matrix that I see are `42` and `true`. Why do you need this large switch statement?

Also, you can initialize the attribute directly in the test cases initialization, and only do `span.SetAttributes(testCase.attr)` here